users/thystips/thystips.c

```diff
@@ -0,0 +1,270 @@
+    if (state->count == 1 || keymap_config.no_gui) {
+        unregister_code16(KC_LSFT);
+    } else {
+        unregister_code(KC_CAPS);
+    }
+}
+
+qk_tap_dance_action_t tap_dance_actions[] = {
+    // Tap once for shift, twice for Caps Lock
+    [TD_LSFT_CAPSLOCK] = ACTION_TAP_DANCE_DOUBLE(KC_LSFT, KC_CAPS),
+    [TD_LSFT_CAPS_WIN] = ACTION_TAP_DANCE_FN_ADVANCED(NULL, dance_LSFT_finished, dance_LSFT_reset),
+};
+#endif // TD_LSFT_CAPSLOCK_ENABLE
+
+// RGB NIGHT MODE
+#ifdef RGB_MATRIX_ENABLE
+static bool rgb_nightmode = false;
+
+// Turn on/off NUM LOCK if current state is different
+void activate_rgb_nightmode(bool turn_on) {
+    if (rgb_nightmode != turn_on) {
+        rgb_nightmode = !rgb_nightmode;
+    }
+}
+
+void set_default_rgb(void) {
+    rgb_matrix_mode(RGB_MATRIX_SOLID_COLOR);
+    rgb_matrix_set_color_all(RGB_NAUTILUS);
+}
+
+bool get_rgb_nightmode(void) {
+    return rgb_nightmode;
+}
+#endif // RGB_MATRIX_ENABLE
+
+// TIMEOUTS
+#ifdef IDLE_TIMEOUT_ENABLE
+static uint16_t timeout_timer     = 0;
+static uint16_t timeout_counter   = 0; // in minute intervals
+static uint16_t timeout_threshold = TIMEOUT_THRESHOLD_DEFAULT;
+
+uint16_t get_timeout_threshold(void) {
+    return timeout_threshold;
+}
+
+void timeout_reset_timer(void) {
+    timeout_timer   = timer_read();
+    timeout_counter = 0;
+};
+
+void timeout_update_threshold(bool increase) {
+    if (increase && timeout_threshold < TIMEOUT_THRESHOLD_MAX) timeout_threshold++;
+    if (!increase && timeout_threshold > 0) timeout_threshold--;
+};
+
+void timeout_tick_timer(void) {
+    if (timeout_threshold > 0) {
+        if (timer_elapsed(timeout_timer) >= 60000) { // 1 minute tick
+            timeout_counter++;
+            timeout_timer = timer_read();
+        }
+#    ifdef RGB_MATRIX_ENABLE
+        if (timeout_threshold > 0 && timeout_counter >= timeout_threshold) {
+            rgb_matrix_disable_noeeprom();
+        }
+#    endif
+    } // timeout_threshold = 0 will disable timeout
+}
+
+#endif // IDLE_TIMEOUT_ENABLE
+
+#if defined(ALTTAB_SCROLL_ENABLE) || defined(IDLE_TIMEOUT_ENABLE) // timer features
+__attribute__((weak)) void matrix_scan_keymap(void) {}
+
+void matrix_scan_user(void) {
+#    ifdef ALTTAB_SCROLL_ENABLE
+    encoder_tick_alttabscroll();
+#    endif
+#    ifdef IDLE_TIMEOUT_ENABLE
+    timeout_tick_timer();
+#    endif
+    matrix_scan_keymap();
+}
+#endif // ALTTAB_SCROLL_ENABLE or IDLE_TIMEOUT_ENABLE
+
+// PROCESS KEY CODES
+__attribute__((weak)) bool process_record_keymap(uint16_t keycode, keyrecord_t *record) {
+    return true;
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    if (!process_record_keymap(keycode, record)) {
+        return false;
+    }
+    switch (keycode) {
+        case KC_00:
+            if (record->event.pressed) {
+                // when keycode KC_00 is pressed
+                SEND_STRING("00");
+            } else
+                unregister_code16(keycode);
```

Comment:
These are not needed, as unregister_code16 doesn't handle them, at all. 

They can/should be removed.